### PR TITLE
Add Jakarta JSON Processing standalone TCK runner

### PR DESCRIPTION
**Pull request by avpinchuk**
To run TCK:

```
mvn clean install -Ptck -pl :glassfish-external-tck-jsonp
```

Results:

```
[INFO] 
[INFO] Results:
[INFO] 
[INFO] Tests run: 179, Failures: 0, Errors: 0, Skipped: 0
[INFO] 
```

### CODE REVIEW

**Comment by pzygielo** [comment]
I'm getting indeed
```
[INFO] Results:
[INFO] 
[INFO] Tests run: 179, Failures: 0, Errors: 0, Skipped: 0
```
but then
```
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  01:19 min
[INFO] Finished at: 2024-04-27T16:03:00+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-surefire-report-plugin:3.2.5:report-only (generate-test-report) on project glassfish-external-tck-jsonp: Execution generate-test-report of goal org.apache.maven.plugins:maven-surefire-report-plugin:3.2.5:report-only failed: Error parsing JUnit XML report appserver/tests/tck/jsonp/target/surefire-reports/TEST-ee.jakarta.tck.jsonp.api.jsongeneratortests.ClientTests.xml: An invalid XML character (Unicode: 0xffff) was found in the CDATA section.
```

```
appserver/tests/tck/jsonp/target/surefire-reports/TEST-ee.jakarta.tck.jsonp.api.jsongeneratortests.ClientTests.xml:266.26: Input is not proper UTF-8, indicate encoding !
Bytes: 0xEF 0xBF 0xBF 0x22: Bytes: 0xEF 0xBF 0xBF 0x22

sonGeneratorObjectTest5 Testing read of {"unicodechars":"\u0000\u000f\u001fÿ＀
                                                                               ^
appserver/tests/tck/jsonp/target/surefire-reports/TEST-ee.jakarta.tck.jsonp.api.jsongeneratortests.ClientTests.xml - invalid
```

(openjdk 21.0.3, Apache Maven 3.9.6)

**Comment by avpinchuk** [comment]
Failed on JDK 18+.

File: appserver/tests/tck/jsonp/pom.xml [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+                        </goals>
+                        <configuration>
+                            <artifactItems>
+                                <artifactItem>
+                                    <groupId>org.glassfish.main.distributions</groupId>
+                                    <artifactId>${glassfish.artifact.id}</artifactId>
+                                    <version>${glassfish.version}</version>
+                                    <type>zip</type>
+                                    <overWrite>false</overWrite>
+                                    <outputDirectory>${glassfish.root}</outputDirectory>
```
Comment (pzygielo) [inline appserver/tests/tck/jsonp/pom.xml:117]:
```
[INFO] --- surefire-report:3.2.5:report-only (generate-test-report) @ glassfish-external-tck-jsonp ---
[WARNING]  Parameter 'outputDirectory' is read-only, must not be used in configuration
```

Comment (pzygielo) [reply appserver/tests/tck/jsonp/pom.xml:117]: not this one :)

File: appserver/tests/tck/jsonp/pom.xml [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+                    <execution>
+                        <id>generate-test-report</id>
+                        <phase>verify</phase>
+                        <goals>
+                            <goal>report-only</goal>
+                        </goals>
+                    </execution>
+                </executions>
+                <configuration>
+                    <outputDirectory>${project.build.directory}/surefire-reports</outputDirectory>
```
Comment (pzygielo) [inline appserver/tests/tck/jsonp/pom.xml:153]:
```
[INFO] --- surefire-report:3.2.5:report-only (generate-test-report) @ glassfish-external-tck-jsonp ---
[WARNING]  Parameter 'outputDirectory' is read-only, must not be used in configuration
```

**Review by pzygielo — DISMISSED** [review]

File: appserver/tests/tck/jsonp/pom.xml [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+            <artifactId>junit-jupiter-api</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.junit.jupiter</groupId>
+            <artifactId>junit-jupiter-engine</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.netbeans.tools</groupId>
```
Comment (arjantijms) [inline appserver/tests/tck/jsonp/pom.xml:91]: Should maybe use the newer jakarta tck version.

```xml
<groupId>jakarta.tck</groupId>
<artifactId>sigtest-maven-plugin</artifactId>
```

Comment (pzygielo) [reply appserver/tests/tck/jsonp/pom.xml:91]: With Jakarta's plugin I'm getting:
```
[ERROR] Errors: 
[ERROR]   JSONPSigTest.signatureTest:281 Fault JSONPSigTest.signatureTest() failed!, diffs found
```

This is, at least, interesting...

Comment (pzygielo) [reply appserver/tests/tck/jsonp/pom.xml:91]: I missed that before - this `sigtest-maven-plugin` is declared as dependency here. Is it used as plugin somewhere?

Comment (avpinchuk) [reply appserver/tests/tck/jsonp/pom.xml:91]: > I missed that before - this `sigtest-maven-plugin` is declared as dependency here. Is it used as plugin somewhere?

Not used. It's called directly from code:

https://github.com/jakartaee/jsonp-api/blob/21733cff9ddf08338710d87d23739c9c3276553f/tck/tck-tests/src/main/java/ee/jakarta/tck/jsonp/signaturetest/SigTestDriver.java#L163

Comment (avpinchuk) [reply appserver/tests/tck/jsonp/pom.xml:91]: > Should maybe use the newer jakarta tck version.

> This is, at least, interesting...

Works with `jakarta.tck:sigtest-maven-plugin:[2.0-2.2]` but failed with `jakarta.tck:sigtest-maven-plugin:2.3`

File: appserver/tests/tck/jsonp/pom.xml (hunk 2 above)
Comment (avpinchuk) [reply appserver/tests/tck/jsonp/pom.xml:153]: I'll try to do something about this warning ;)

**Review by arjantijms — APPROVED** [review]

**Pull request merged**